tools/layer-zero-example/contracts/HTSConnector.sol

```diff
@@ -0,0 +1,122 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.20;
+
+import {OFTCore} from "@layerzerolabs/lz-evm-oapp-v2/contracts/oft/OFTCore.sol";
+import "./hts/HederaTokenService.sol";
+import "./hts/IHederaTokenService.sol";
+import "./hts/KeyHelper.sol";
+
+/**
+ * @title HTS Connector
+ * @dev HTS Connector is a HTS token that extends the functionality of the OFTCore contract.
+ */
+abstract contract HTSConnector is OFTCore, KeyHelper, HederaTokenService {
+    address public htsTokenAddress;
+
+    event TokenCreated(address tokenAddress);
+
+    /**
+     * @dev Constructor for the HTS Connector contract.
+     * @param _name The name of HTS token
+     * @param _symbol The symbol of HTS token
+     * @param _lzEndpoint The LayerZero endpoint address.
+     * @param _delegate The delegate capable of making OApp configurations inside of the endpoint.
+     */
+    constructor(
+        string memory _name,
+        string memory _symbol,
+        address _lzEndpoint,
+        address _delegate
+    ) payable OFTCore(8, _lzEndpoint, _delegate) {
+        IHederaTokenService.TokenKey[] memory keys = new IHederaTokenService.TokenKey[](2);
+        keys[0] = getSingleKey(
+            KeyType.ADMIN,
+            KeyType.PAUSE,
```

Comment:
I don't think we need the pause key. Let's keep this part simple with just admin and supply.